File: src/main/java/net/yogstation/yogbot/util/ChannelUtil.java
package net.yogstation.yogbot.util;

import net.yogstation.yogbot.Yogbot;
import net.yogstation.yogbot.config.DiscordChannelsConfig;

import java.util.Arrays;
import java.util.HashSet;
import java.util.Set;

public class ChannelUtil {
	public static boolean isAdminChannel(long channelID) {
		DiscordChannelsConfig config = Yogbot.config.channelsConfig;
		final Set<Long> adminChannelIds = new HashSet<>(Arrays.asList(
			config.channelAdmemes,
				config.channelAdmin,
				config.channelCouncil,
				config.channelAdminBotspam
		));
		return adminChannelIds.contains(channelID);
	}
}

File: src/main/java/net/yogstation/yogbot/util/ChannelUtil.kt
package net.yogstation.yogbot.util

import net.yogstation.yogbot.config.DiscordChannelsConfig
import java.util.HashSet
import java.util.Arrays

object ChannelUtil {
	fun isAdminChannel(channelID: Long, config: DiscordChannelsConfig): Boolean {
		val adminChannelIds: Set<Long> = HashSet(
			listOf(
				config.channelAdmemes, config.channelAdmin, config.channelCouncil,
				config.channelAdminBotspam
			)
		)
		return adminChannelIds.contains(channelID)
	}
}
